File: sdk/client/javascript/openapi-test/test/klaytnDebug/others/DebugSetMutexProfileFraction.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('debug_setMutexProfileFraction API', () => {
    test('should return debug_setMutexProfileFraction', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data.jsonrpc).toBe("2.0");
            expect(data.result).toBeDefined()
            done();
        };

        const rate = 2;

        sdk.debug.setMutexProfileFraction(rate, {}, callbackOne);
    });
});

File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/klaytnDebug/others/DebugSetMutexProfileFractionTest.java
package opensdk.sdk.apis.klaytnDebug.others;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugSetMutexProfileFractionResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

@DisplayName("Debug RPC Test")
public class DebugSetMutexProfileFractionTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC debug_setMutexProfileFraction")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        int rate = 2;

        DebugSetMutexProfileFractionResponse response = sdk.debug.setMutexProfileFraction(rate).send();

        assertNotNull(response);
        assertNull(response.getError());
    }
}
